### DISCUSSION TITLE
Why does Netbox displays duplicated IP addresses when theses IPs are in differents VLANs ?

### DISCUSSION THREAD

**Discussion by Wait4Code** [netbox-community/netbox#12710] [Q&A] [answered]
I am using a Nutanix virtualization platform and VMs networks are separated using VLANs. Private IPS are therefore widely reused inside my clusters and when i'm trying to describe this inside Netbox, I am a bit annoyed that IP appears has "duplicated" whereas they are affected to interfaces that have differents VLANs.

Is there a way to prevent this ? What am I doing wrong ?

**Comment by bogdancordos** [comment]
You can reuse the same IP in different VRFs, so you might want to look into segmenting your cluster to use multiple VRFs. This can help: https://demo.netbox.dev/static/docs/models/ipam/vrf/

**Comment by Wait4Code** [comment]
Well, I know that I can use same IPs in different VRFs and in fact I am using them in other parts of my datacenters but in my virtualization cluster I use VLANs segmentation. 

I have neither the power nor the will to change my infrastructure to match Netbox model, I was just wondering if there was a way to describe my cluster accurately but maybe I should open a feature request ?

**Reply by candlerb** [reply, marked as answer]
If you have two different VLANs using the same prefix, then they can't be part of the same routing table, i.e. they must be separate at layer 3.

In Netbox the concept of "VRF" is also mixed with the idea of "Network namespace", i.e. a uniqueness zone for IP addresses.

This doesn't mean you need to have actual VRFs in your network, but you can still model these networks linked to "VRFs" in Netbox to document that they are distinct uses of the same IP block.  I've used this for the private point-to-point links between firewalls in a cluster, for example.
